src/pyramid_jwt/policy.py

```diff
@@ -94,3 +98,127 @@ def forget(self, request):
             'has no effect.',
             stacklevel=3)
         return []
+
+
+class ReissueError(Exception):
+    pass
+
+
+@implementer(IAuthenticationPolicy)
+class JWTTokenAuthenticationPolicy(JWTAuthenticationPolicy):
+    def __init__(self, private_key, public_key=None, algorithm='HS512',
+                 leeway=0, expiration=None, default_claims=None,
+                 http_header='Authorization', auth_type='JWT',
+                 callback=None, json_encoder=None, audience=None,
+                 cookie_name='Authorization', https_only=False,
+                 reissue_time=None):
+        super(JWTTokenAuthenticationPolicy, self).__init__(
+            private_key, public_key, algorithm,
+            leeway, expiration, default_claims,
+            http_header, auth_type,
+            callback, json_encoder, audience)
+
+        self.https_only = https_only
+        self.cookie_name = cookie_name
+        self.max_age = self.expiration and self.expiration.total_seconds()
+
+        if reissue_time and isinstance(reissue_time, datetime.timedelta):
+            reissue_time = reissue_time.total_seconds()
+        self.reissue_time = reissue_time
+
+        self.cookie_profile = CookieProfile(
+            cookie_name=self.cookie_name,
+            secure=self.https_only,
+            max_age=self.max_age,
+            httponly=True,
+            path=None
+        )
+
+    @staticmethod
+    def make_from(policy, **kwargs):
+        if not isinstance(policy, JWTAuthenticationPolicy):
+            pol_type = policy.__class__.__name__
+            raise AttributeError('Invalid policy type %s' % pol_type)
+
+        return JWTTokenAuthenticationPolicy(
+            private_key=policy.private_key,
+            public_key=policy.public_key,
+            algorithm=policy.algorithm,
+            leeway=policy.leeway,
+            expiration=policy.expiration,
+            default_claims=policy.default_claims,
+            http_header=policy.http_header,
+            auth_type=policy.auth_type,
+            callback=policy.callback,
+            json_encoder=policy.json_encoder,
+            audience=policy.audience,
+            **kwargs
+        )
+
+    def _get_cookies(self, request, value, max_age=None):
+        profile = self.cookie_profile(request)
```

Comment:
Is there a reason to explicitly set the domain? This already defaults to the current host, which is the safest option.

Reply:
It is not. The CookieProfile will default to no domain if it's not instantiated with one.

Note that at the time of creating the CookieProfile we don't have a request, so we don't know the current domain. On top of that, the user may want to add multiple domains to this cookie. I'll refactor to allow user override.